code/modules/awaymissions/mission_code/ruins/deepstorage.dm

```diff
@@ -0,0 +1,380 @@
+	// is our mob dead?
+	var/has_died = FALSE
+	loot = list(/obj/item/gun/energy/laser,
+			/obj/effect/decal/cleanable/blood/innards,
+			/obj/effect/decal/cleanable/blood,
+			/obj/effect/gibspawner/generic,
+			/obj/effect/gibspawner/generic) // first weapon this ruin provides
+
+/mob/living/simple_animal/hostile/spaceinfected/gateopener/Destroy()
+	handle_dying()
+	return ..()
+
+/mob/living/simple_animal/hostile/spaceinfected/gateopener/proc/handle_dying()
+	if(!has_died)
+		has_died = TRUE
+
+/mob/living/simple_animal/hostile/spaceinfected/gateopener/death(gibbed)
+	if(can_die() && !has_died)
+		unlock_blast_doors(DS_ENGINEERING)
+	return ..(gibbed)
+
+/mob/living/simple_animal/hostile/spaceinfected/gateopener/proc/unlock_blast_doors(target_id_tag)
+	for(var/obj/machinery/door/poddoor/P in GLOB.airlocks)
+		if(P.density && P.id_tag == target_id_tag && P.z == z && !P.operating)
+			P.open()
+
+/mob/living/simple_animal/hostile/spaceinfected/default/ranged
+	desc = "A reanimated corpse. This one is keeping its distance from you."
+	icon_state = "spaceinfected_ranged"
+	ranged = TRUE
+	retreat_distance = 5
+	minimum_distance = 5
+	projectiletype = /obj/item/projectile/neurotox
+	projectilesound = 'sound/weapons/pierce.ogg'
+
+// Below here is ruin specific code
+
+/obj/structure/blob/normal/deepstorage //parent one didn't allow me to color them with varedit
+	name = "flesh wall"
+	desc = "What even..."
+	color = rgb(80, 39, 39)
+
+/obj/machinery/deepstorage_teleporter
+	name = "package teleporter"
+	desc = "It's tuned to maintain one-way teleportation."
+	icon = 'icons/obj/stationobjs.dmi'
+	icon_state = "controller"
+	density = TRUE
+	anchored = TRUE
+
+	// how many portals are present right now?
+	var/active_portals = 0
+	// do user has the tgui menu open right now?
+	var/menu_open = FALSE
+
+/obj/machinery/deepstorage_teleporter/attack_hand(mob/user)
+	if(active_portals != 0 || menu_open)
+		return
+	menu_open = TRUE
+	var/list/boss_warning = list("Proceed" = TRUE)
+	var/final_decision = tgui_input_list(user, "Just a hunch but wherever this machine may lead, it won't be somewhere so pleasant. Are you sure about this?", "Make your decision", boss_warning)
+	if(!final_decision)
+		to_chat(user, "<span class='notice'>The teleporter machine remains untouched.</span>")
+		menu_open = FALSE
+		return
+
+	new /obj/effect/portal/advanced/deepstorage(locate(x, y + 1, z), locate(x + 3, y- 6, z), src, 200)
+	playsound(loc, 'sound/machines/twobeep.ogg', 50, TRUE)
+	active_portals++
+	addtimer(CALLBACK(src, PROC_REF(cooldown_passed)), 20 SECONDS)
+	menu_open = FALSE
+
+// this proc is called when portal disappears after a while, so users can interact with teleporter again
+/obj/machinery/deepstorage_teleporter/proc/cooldown_passed()
+	active_portals--
+
+/obj/effect/portal/advanced/deepstorage
+	name = "portal"
+	desc = "Good luck."
+
+// paper stuff & lore
```

Comment:
I really love all the work you put into the lore here, it's actually kind of creepy. I've made a lot of syntax/grammar suggestions and reworded some parts to make them easier to follow as a native English speaker.

Reply:
thank you for putting this mess in an order